kore/src/Kore/Step/Simplification/Unify.hs

```diff
@@ -0,0 +1,1006 @@
+    , matchFirstBottom
+    , matchFirstElemVar
+    , matchInjApp
+    , matchInjs
+    , matchInt
+    , matchString
+    , matchStringLiteral
+    , matchUnDefined
+    , matchUnifyBoolAnd1
+    , matchUnifyBoolAnd2
+    , matchUnifyBoolNot
+    , matchUnifyBoolOr1
+    , matchUnifyBoolOr2
+    , matchUnifyKequalsEq
+    , matchUnifyIntEq
+    , matchUnifyStringEq
+    , matchUnifyEqualsEndianness
+    , matchUnifyEqualsSignedness
+    , matchUnifyNotInKeys
+    , matchDomainValueAndConstructorErrors1
+    , matchDomainValueAndConstructorErrors2
+    , matchUnifyOverload1
+    , matchUnifyOverload2
+    , matchUnifyOverload3
+    , matchUnifyOverload4
+    , matchUnifyOverload5
+    , matchUnifyOverload6
+    , matchUnifyEqualsMap
+    , matchUnifyEqualsSet
+    , matchUnifyEqualsList
+    , matchBoolAnd1
+    , matchBoolAnd2
+    , matchBoolAnd3
+    , matchBoolAnd4
+    , matchVariableFunctionAnd1
+    , matchVariableFunctionAnd2
+    , matchITE
+    , matchExpandAlias
+    , matchIsFunctionPatterns
+    , Unification (..)
+    , UnifyBoolOrArgs (..)
+    , UnifyBoolNotArgs (..)
+    , UnifyKequalsEqArgs (..)
+    , UnifyIntEqArgs (..)
+    , UnifyStringEqArgs (..)
+    , UnifyEqualsEndiannessArgs (..)
+    , UnifyEqualsSignednessArgs (..)
+    , UnifyNotInKeysArgs (..)
+    , UnifyOverload1Args (..)
+    , UnifyOverload2Args (..)
+    , UnifyOverload3Args (..)
+    , UnifyOverload4Args (..)
+    , UnifyOverload5Args (..)
+    , UnifyOverload6Args (..)
+    , UnifyEqualsList3Args (..)
+    , UnifyEqualsList4Args (..)
+    , UnifyEqualsList5Args (..)
+    , UnifyMapEqualsArgs (..)
+    , UnifyMapEqualsVarArgs (..)
+    , UnifySetEqualsArgs (..)
+    , UnifySetEqualsVarArgs (..)
+    , UnifyExpandAliasArgs (..)
+    ) where
+
+import qualified Data.Functor.Foldable as Recursive
+import qualified Data.List as List
+import qualified Data.Map.Strict as M
+import Data.Text
+-- import Pretty (
+--     Doc
+--  )
+
+import qualified Kore.Attribute.Symbol as Symbol
+import qualified Kore.Builtin.AssociativeCommutative as Ac
+import qualified Kore.Builtin.Bool as Bool
+import qualified Kore.Builtin.Builtin as Builtin
+import Kore.Builtin.Endianness
+import Kore.Builtin.EqTerm
+import qualified Kore.Builtin.Int as Int
+import Kore.Builtin.KEqual
+import qualified Kore.Builtin.List.List as List
+import qualified Kore.Builtin.Map as Map
+import Kore.Builtin.Signedness
+import qualified Kore.Builtin.String as String
+import Kore.IndexedModule.MetadataTools
+import Kore.Internal.InternalBool
+import Kore.Internal.InternalInt
+import Kore.Internal.InternalList
+import Kore.Internal.InternalMap
+import Kore.Internal.InternalSet
+import Kore.Internal.InternalString
+import Kore.Internal.Symbol
+import Kore.Internal.TermLike
+import Kore.Rewriting.RewritingVariable
+import Kore.Step.Simplification.ExpandAlias
+import Kore.Syntax.PatternF
+    ( Const (..)
+    )
+import Kore.TopBottom
+import Prelude.Kore
+
+data Unification
```

Comment:
This shouldn't exist. These types shouldn't be tied together. For each `match...` and `unify...` function, there should be one data type (unique to that pair) that goes between them. The unique data type and the `match...` function should be in the same module as the `unify...` function.